emgapianns/management/commands/import_run.py

```diff
@@ -40,73 +40,68 @@ class Command(BaseCommand):
         parser.add_argument('--library_strategy', help='Library strategy',
                             choices=['AMPLICON', 'WGS', 'ASSEMBLY', 'RNA-Seq', 'WGA'])
 
     def handle(self, *args, **options):
         logger.info("CLI %r" % options)
         self.emg_db = options['emg_db']
-        self.ena_db = options['ena_db']
+        self.import_study = options['import_study']
         if options['result_dir']:
             self.result_dir = os.path.abspath(options['result_dir'])
         if options['biome']:
             self.biome = options['biome']
         if options['library_strategy']:
             self.library_strategy = options['library_strategy']
-
         for acc in options['accessions']:
             logger.info('Importing run {}'.format(acc))
             self.import_run(acc)
             logger.info("Run import finished successfully.")
 
     def import_run(self, accession):
         api_run_data = self.get_run_api(accession)
-        db_run_data = self.get_ena_db_run(accession)
-        run = self.create_or_update_run(db_run_data, api_run_data)
-        self.tag_study(run, api_run_data['secondary_study_accession'])
-        self.tag_sample(run, api_run_data['secondary_sample_accession'])
+        run = self.create_or_update_run(api_run_data)
+        
         if self.library_strategy:
             _library_strategy = self.library_strategy
         else:
             _library_strategy = api_run_data['library_strategy']
         _library_source = api_run_data['library_source']
         self.tag_experiment_type(run, identify(library_strategy=_library_strategy,
                                                library_source=_library_source).value)
+
+        self.tag_study(run, api_run_data['secondary_study_accession'], get_or_create=self.import_study)
+        self.tag_sample(run, api_run_data['secondary_sample_accession'])
+
+        # Validate and save #
+        run.full_clean()
         run.save(using=self.emg_db)
 
     @staticmethod
     def get_run_api(accession):
         logger.info('Fetching run {} from ena api'.format(accession))
         return ena.get_run(accession)
 
-    def get_ena_db_run(self, accession):
-        logger.info('Fetching run {} from ena oracle DB'.format(accession))
-        return ena_models.Run.objects.using(self.ena_db).filter(run_id=accession)[0]
-
-    def create_or_update_run(self, db_data, api_data):
+    def create_or_update_run(self, api_data):
         accession = api_data['run_accession']
-        logger.info('Creating run {}'.format(accession))
-        status = emg_models.Status.objects.using(self.emg_db).get(status_id=db_data.status_id)
+        logger.info(f'Creating run {accession}')
         defaults = utils.sanitise_fields({
             'instrument_platform': api_data['instrument_platform'],
             'instrument_model': api_data['instrument_model'],
-            'status_id': status,
+            'status_id_id': int(api_data["status_id"]),
```

Comment:
Is this a typo or supposed to hint at something?

Reply:
good catch.. that column is called status_id even though is a FK :face_exhaling: